haystack/components/converters/docx.py

```diff
@@ -89,56 +128,37 @@ def run(
                 )
                 continue
 
-            # Load the Metadata
-            try:
-                docx_meta = self._get_docx_metadata(document=file)
-            except Exception as e:
-                logger.warning(
-                    "Could not load the metadata from {source}, skipping. Error: {error}", source=source, error=e
-                )
-                docx_meta = {}
-
-            # Load the content
-            try:
-                paragraphs = [para.text for para in file.paragraphs]
-                text = "\n".join(paragraphs)
-            except Exception as e:
-                logger.warning(
-                    "Could not convert {source} to a Document, skipping it. Error: {error}", source=source, error=e
-                )
-                continue
-
-            merged_metadata = {**bytestream.meta, **docx_meta, **metadata}
+            docx_metadata = self._get_docx_metadata(document=file)
+            merged_metadata = {**bytestream.meta, **metadata, "docx": docx_metadata}
             document = Document(content=text, meta=merged_metadata)
-
             documents.append(document)
 
         return {"documents": documents}
 
-    def _get_docx_metadata(self, document: "DocxDocument") -> Dict[str, Union[str, int, datetime]]:
+    def _get_docx_metadata(self, document: "DocxDocument") -> DocxMetadata:
         """
         Get all relevant data from the 'core_properties' attribute from a Docx Document.
 
         :param document:
             The Docx Document you want to extract metadata from
 
         :returns:
-            A dictionary containing all the relevant fields from the 'core_properties'
+            A `DocxMetadata` dataclass all the relevant fields from the 'core_properties'
         """
-        return {
-            "author": document.core_properties.author,
-            "category": document.core_properties.category,
-            "comments": document.core_properties.comments,
-            "content_status": document.core_properties.content_status,
-            "created": document.core_properties.created,
-            "identifier": document.core_properties.identifier,
-            "keywords": document.core_properties.keywords,
-            "language": document.core_properties.language,
-            "last_modified_by": document.core_properties.last_modified_by,
-            "last_printed": document.core_properties.last_printed,
-            "modified": document.core_properties.modified,
-            "revision": document.core_properties.revision,
-            "subject": document.core_properties.subject,
-            "title": document.core_properties.title,
-            "version": document.core_properties.version,
-        }
+        return DocxMetadata(
+            author=document.core_properties.author,
+            category=document.core_properties.category,
+            comments=document.core_properties.comments,
```

Comment:
Hey @shadeMe I had a question about this extraction. All of the values in `core_properties` are `@property` types and from what I can tell in their source code always are defined with default values. So I wanted to ask should we be extra careful here and still check if the attribute is present using a `if hasattr()`, or would this be fine to leave?

Reply:
Maybe I'm misunderstanding the question, but it should be fine if they are properties? Why does it matter if they return a default value or not?

Reply:
I was more worried they'd throw an attribute error. I guess being a property doesn't guarantee that no attribute error is thrown depending on how the getter looks. So I was mostly checking their source code to make sure the getter accessed a variable that actually existed. Otherwise I think checking if the attribute exists first would be a good idea.

Reply:
It looks safe and behaves safe enough when I created an empty docx doc everything was defined. It was just hard to follow their source to 100% certain. But I'll go ahead and leave it.